packages/xo-server/src/xo-mixins/remotes.js

```diff
@@ -142,22 +153,30 @@ export default class {
     }
 
     return this._updateRemote(id, {
+      benchmarks,
+      enabled,
       name,
-      url,
       options,
-      enabled,
+      url,
     })
   }
 
   @synchronized()
-  async _updateRemote(id, { url, ...props }) {
+  async _updateRemote(id, { benchmarks, url, ...props }) {
     const remote = await this._getRemote(id)
 
     // url is handled separately to take care of obfuscated values
     if (typeof url === 'string') {
       remote.url = format(sensitiveValues.merge(parse(url), parse(remote.url)))
     }
 
+    if (benchmarks !== undefined) {
+      remote.benchmarks = JSON.stringify(benchmarks)
+    }
+    if (Array.isArray(remote.benchmarks) === true) {
```

Comment:
What's the case which it will be an array?

Reply:
`_updateRemote` is called many times for refresh data i believe. Data in remote object are `JSON.parse()`, so we have to stringify them to avoid issues.

Reply:
Right, then i suggest:
```js
if (benchmarks !== undefined || (benchmarks = remote.benchmarks) !== undefined)
```